codec.go

```diff
@@ -57,6 +60,7 @@ func (e *framedMsgpackEncoder) Encode(i interface{}) error {
 		return err
 	}
 	e.writeCh <- bytes
+	// FIXME race here?
```

Comment:
I had thought of this scenario, but wasn't 100% sure whether it can race:. In retrospect I don't think it can.
* G1 Encode
* G2 Encode
Now they are both waiting on e.resultCh. I was afraid that there might be a race as to which one gets returned first, but I don't think so any more. The ordering is preserved because there's only one place each channel is read and written.